### DISCUSSION TITLE
Http Client

### DISCUSSION THREAD

**Discussion by ZacHooper** [saviorand/lightbug_http#23] [Q&A] [answered]
Is there scope for this project to also be the Client side of HTTP or is mainly focused on the server side?

**Comment by saviorand** [comment]
@ZacHooper there is a (very minimal) python-based client implementation available. See a test for it here https://github.com/saviorand/lightbug_http/blob/1eb9242ce0ddeeec39ac858028a7117dde627523/lightbug_http/tests/test_client.mojo#L13 , just updated it to work in the latest version of Mojo.
I'm definitely interested in adding a version based on C external_call's that doesn't invoke Python, so only Mojo-based, like in the server! Would love contributions here

**Reply by ZacHooper** [reply]
Awesome, will have a play with the Python version. I've been tinkering around with the C external_call's already (still very early days) so may be able to contribute a little 😄

**Comment by saviorand** [comment]
For those who read this: there is now a pure Mojo implementation in sys/client (no Python involved) that leverages external calls to C, kudos to @SchmittB and @ZacHooper ! 
This was added in https://github.com/saviorand/lightbug_http/pull/26 and https://github.com/saviorand/lightbug_http/pull/28

**Comment by saviorand** [comment, marked as answer]
There's a section in the README on using the client now: https://github.com/saviorand/lightbug_http?tab=readme-ov-file#using-the-client
